src/main/java/edu/umich/ctools/esb/utils/WAPI.java

```diff
@@ -0,0 +1,213 @@
+	private final static String SCOPE = "scope";
+	private final static String PRODUCTION = "PRODUCTION";
+	private final static String ERROR_MSG = "No results due to error. See meta for more details.";
+
+	private final String BEARER = "Bearer";
+
+	private String apiPrefix;
+	private String tokenServer;
+	private String key;
+	private String secret;
+	private String renewal;
+	private String token;
+	
+	public WAPI() {
+		super();
+	}
+
+	//WAPI constructor will have single variable which will be a map holding all necessary variables
+	//The values for the map should come from a properties file used by the application that depends
+	//on this library. For example, if ccm uses the esb-utils library, the ccm properties should
+	//contain the values for the map being passed in to the WAPI constructor.
+	public WAPI(HashMap<String, String> value) {
+		this.setApiPrefix(value.get("apiPrefix"));
+		this.tokenServer = value.get("tokenServer");
+		this.key = value.get("key");
+		this.secret = value.get("secret");
+		this.renewal = buidRenewal(this.key, this.secret);
+		
+		M_log.info("tokenServer: " + tokenServer);
+		M_log.info("key: " + this.key);
+		M_log.info("secret: " + this.secret);
+		M_log.info("renewal: " + this.renewal);
+		M_log.info("token: " + this.token);
+	}
+	
+	public String getApiPrefix() {
+		return apiPrefix;
+	}
+
+	public void setApiPrefix(String apiPrefix) {
+		this.apiPrefix = apiPrefix;
+	}
+	
+	//a token must be created at the time of construction
+	//this token will allow use of the ESB APIs
+	public String buidRenewal(String key, String secret) {
+		String b64 = base64KeySecret(key, secret);
+		b64 = "Basic " + b64; 
+		return b64;
+	}
+
+	//esb calls require base 64 strings for authroization
+	public String base64KeySecret(String key, String secret) {
+		String keySecret = key + ":" + secret;
+		byte[] binaryData = keySecret.getBytes();
+		keySecret = Base64.encodeBase64String(binaryData);
+		return keySecret;
+	}
+
+	//perform ESB request. If there is an exception, return exception result response.
+	public WAPIResultWrapper doRequest(String request) throws UnirestException{
+		M_log.info("doRequest: " + request);
+		WAPIResultWrapper wrappedResult = null;
+		JSONObject jsonObject = null;
+		HttpResponse<String> response = null;
+		try{
+			response = Unirest.get(request)
+					.header(AUTHORIZATION, this.token)
+					.header("Accept", "json")
+					.asString();
+			M_log.info("Raw body: " + response.getBody());
+			M_log.info("Status: " + response.getStatus());
+			M_log.info("Status Text: " + response.getStatusText());
+			jsonObject = new JSONObject(response.getBody());
+			wrappedResult = new WAPIResultWrapper(response.getStatus(), "COMPLETED", jsonObject);
+		}
+		catch(Exception e){
+			M_log.error("Error in doRequest: " + e.getMessage());
+			wrappedResult = reportError(response.getStatus());
+		}
+		return wrappedResult;
+	}	
+	
+	//Error handling for bad calls
+	public WAPIResultWrapper reportError(int status) {
```

Comment:
Can more be included than just the status?  Some errors might have an error message or a body.